### PR TITLE
Release v4.0.0

### LINKED ISSUE #199: Add examples for different workflows
Look around for projects that use those workflows and add some expamples to show during the training

### LINKED ISSUE #200: Make a lab for building a release
Maybe as an addition to the CI / CD chapter to have the html be uploaded into the release

https://docs.gitlab.com/ee/user/project/releases/

### LINKED ISSUE #209: Restructure the Gitflow Workflow slides
I think it would make more sense to add more focus on what types of branches serve which purpose, instead of explaining a strict gitflow?
We already sort of have that in the slides, but right now it feels like we are explaining a system with its components, rather than showing the components to be freely assembled into what fits your needs.
What do you think @martialblog ?

In addition we could also add some information about support branches ("contains older stable versions that still need to receive critical bug fixes but no features") - as that would be lost when we remove the two example project slides?
See my last comment in #199

### LINKED ISSUE #214: Evaluate using a single NWS account for providing Gitlab Instances
A suggestion from the last internal training: There seems to be a possibility to provide multiple instances of Gitlab using only a single account. I haven't tried this and I'm not too sure how the exact workflow for this looks like - we should evaluate this.

### LINKED ISSUE #216: Idea: Remove day2/02_Integrations and use the time for more CI
I think the day2/02_Integrations chapter is not that beneficial for the training, we could skip it and use the extra time for more GitLab topics. Most people are curious about CI, so that would be a good topic to extend.

### LINKED ISSUE #217: Invite the trainer to your project
Should be around Lab 6.3

### LINKED ISSUE #220: Revert the graphics for the workflows chapters
They were a little less confusing and more clear than the mermaids

### PR DESCRIPTION
**Pull request by martialblog**
Major Refactor of the Training. 

Day 1:

- Smoother Start into Git core concepts
- Add more GitLab Topics on day1
- Add more GitLab Labs on day1
- Reword a lot of slides to be clearer
- Remove Create NMS GitLab Labs (as discussed)
- Move git tag to end of day and finish the day with a git tag v0.5 -m "End of Day One Release"

Day 2:

- Remove Integrations Chapter on day2, time will be used for more CI
- Optimize CI Chapter and add new CI Labs
- Streamline Git Workflow Examples
- Optimize rebase and squashing Slides
- Reword a lot of slides to be clearer
- Reword Git Workflow Slides and revert GitFlow images
- Add Release Pipeline Lab with Release v1.0 as "End of Training Release"

Fixes #209 

Fixes #220 

Fixes #200 

Fixes #217 

Fixes #214 

Fixes #199 

Fixes #216

### CODE REVIEW

File: showoff.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "GitLab",
   "subtitle": "Version Control System & Continuous Integration",
   "location": "Nuremberg",
-  "author": "Feu Mourek, Henrik Triem, Markus Opolka, <br><small>originally Michael Friedrich</small>",
```
Comment (martialblog) [inline showoff.json:5]: :+1:

File: day1/06_Server/04_Connect_Local_Remote.md [hunk 2]
```diff
@@ -346,9 +397,8 @@ This is only for the training. In reality use SSH authentication.
 
 Choose `History` and look at the Git commits, their author, subject and timestamp.
 
-Compare it with the local `git log` or `tig` entries. 
+Compare it with the local `git log` or `tig` entries.
```
Comment (mocdaniel) [inline day1/06_Server/04_Connect_Local_Remote.md:400]:
```suggestion
Compare it to the local `git log` or `tig` entries.
```

File: day2/01_Workflows/04_Gitflow.md [hunk 3]
```diff
@@ -1,114 +1,89 @@
...
-~~~ENDSECTION~~~
+<center><img src="../../_images/git_gitflow_workflow_01.png" alt="Gitflow Workflow" width="75%" /></center>
 
 !SLIDE smbullets noprint
 # Gitflow Workflow - Feature Branches
 
-* New features in their own branches
-* Feature branches use `develop` as their parent branch
-* Once completed, merged back to `develop`
+Development happends on features branches.
```
Comment (mocdaniel) [inline day2/01_Workflows/04_Gitflow.md:33]:
```suggestion
Development happens on feature branches.
```

File: day2/01_Workflows/04_Gitflow.md [hunk 4]
```diff
@@ -1,114 +1,89 @@
...
 # Gitflow Workflow - Feature Branches
 
-* New features in their own branches
-* Feature branches use `develop` as their parent branch
-* Once completed, merged back to `develop`
-
-<center><img src="../../_images/mermaid/workflow-feature-branches.mmd.png" alt="Gitflow Workflow" width="85%" /></center>
-
-~~~SECTION:handouts~~~
+Development happends on features branches.
```
Comment (mocdaniel) [inline day2/01_Workflows/04_Gitflow.md:43]:
```suggestion
Development happens on feature branches.
```

File: day2/03_CI/01_Intro.md [hunk 5]
```diff
@@ -1,22 +1,12 @@
...
-
-~~~SECTION:handouts~~~
-
-****
-
-
-~~~ENDSECTION~~~
+# What is Continuous Integration?
 
+Continuous integration (CI) is the practice to integrating new code more frequently and securely throughout the development.
```
Comment (mocdaniel) [inline day2/03_CI/01_Intro.md:7]:
```suggestion
Continuous integration (CI) is the practice of integrating new code more frequently and securely throughout development.
```

File: day2/03_CI/03_Gitlab_CI.md [hunk 6]
```diff
@@ -81,56 +53,36 @@ https://docs.gitlab.com/runner/register/index.html
...
 
-
 !SLIDE smbullets
-# GitLab CI: Container Registry
+# GitLab CI Configuration
 
-* Enable the Container Registry (administration server setting)
-* Enable the Container Registry for the project
-* Advanced usage only
+The CI is configured by the `.gitlab-ci.yml` in our Git repository
```
Comment (mocdaniel) [inline day2/03_CI/03_Gitlab_CI.md:59]:
```suggestion
The CI is configured by the `.gitlab-ci.yml` file in our Git repository
```

File: day2/03_CI/04_Gitlab_Pipelines.md [hunk 7]
```diff
@@ -159,75 +153,170 @@ This is an example of how to do it from a CLI, the Gitlab WebIDE is an obvious a
...
+# GitLab Pipelines: Triggers and Rules
 
-* Maximum efficiency
-* Disregards stage order
-* Use `needs` keyword to define dependencies between jobs
+We can use the `trigger` keyword to start other pipelines:
+
+* Multi-project pipeline or Child pipeline
+
+We can use the `rules` keyword include or exclude jobs.
```
Comment (mocdaniel) [inline day2/03_CI/04_Gitlab_Pipelines.md:194]:
```suggestion
We can use the `rules` keyword to include or exclude jobs.
```

File: day2/05_Outlook/31_LFS.md [hunk 8]
```diff
@@ -1,9 +1,11 @@
...
-# Git Large File Storage (LFS)
+# Git - Large File Storage (LFS)
 
-* Open source Git extension for versioning large files
-* Replaces large files (audio, videos, datasets, graphics) with text pointers
-* Requires a '.gitattributes' file for tracking files in your repository
+Git LFS replaces large files (audio, videos, datasets, graphics) with text pointers.
+
+* It is generally not recommended to store binary files in Git
+* Git LFS is a plugin to overcome this shortcomming
```
Comment (mocdaniel) [inline day2/05_Outlook/31_LFS.md:7]:
```suggestion
* Git LFS is a plugin to overcome this shortcoming
```

**Review by mocdaniel — CHANGES_REQUESTED** [review]
Looks good, marked some typos here and there, will update the PR.

**Comment by martialblog** [comment]
Rebased/Squashed some commits. @theFeu also confirmed the PR.

Ship it :shipit:

**Pull request merged**